### PR TITLE
add performance doc

### PR DESCRIPTION
**Pull request by samjsong**
###### Description

Based on Vaibhav's performance tests.

I wasn't sure where to link to this doc, just adding it for now. Anybody have any suggestions? :)

### CODE REVIEW

**Review by vsinghal13 — COMMENTED** [review]
Thanks, Sam! One suggestion would be to add the per fluentd replica capacity in terms of gb/day or DPM or provide some sort of formula for the same. This will help the customers get a better idea for the number of fluentd's required according to their load.

**Review by frankreno — APPROVED** [review]
LGTM! Same feedback on including the per replica load so users can easily apply to their loads. Also I wanna make this document more discoverable.  I am thinking on each of the various installation, helm, non-helm side-by-side, etc, it would be good to have some reference here.  Even a simple note. thoughts?

**Comment by samjsong** [comment]
Thanks Frank! That's exactly what I was looking for when I asked about where to link this doc 🙂 I've gone ahead and addressed comments, thanks team

**Review by sumo-drosiek — APPROVED** [review]
I like it 👍

File: deploy/docs/Performance.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    ```
+    - url: http://$(CHART).$(NAMESPACE).svc.cluster.local:9888/prometheus.metrics.node
+      writeRelabelConfigs:
+      - action: keep
+        regex: node-exporter;(?:node_load1|node_load5|node_load15|node_cpu_seconds_total)
+        sourceLabels: [job, __name__]
+      remoteTimeout: 1s
+    ```
+5. For clusters with 2000 application pods, we found that the **Fluentd-events** pod had to be given a 1 GiB memory limit to accommodate the increased events load. If you find that the **Fluentd-events** pod is being OOMKilled, please increase the memory limits and requests accordingly.
+6. For our log generating test application pods, we found that changing the IOPS to 375 or 125GiB improved stability.
```
Comment (pmalek-sumo) [inline deploy/docs/Performance.md:22]: This is not super clear to me:

> ... we found that changing the **IOPS to 375 or 125GiB** improved ... 

changing IOPS to 375 makes sense but to 125GiB? Isn't there a word missing by any chance?

File: deploy/docs/Performance.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Up to 500 application pods
```
Comment (pmalek-sumo) [inline deploy/docs/Performance.md:24]: nitpicks:

* ideally there should be an empty line directly below a heading, otherwise a [MD022](https://github.com/DavidAnson/markdownlint/blob/v0.20.4/doc/Rules.md#md022) is triggered when using `markdown` lint :)
* `Recommendations` section has `##` - the same as this here. IMHO `Up to 500 application pods` should be nested underneath `Recommendations` with `###` (same for `Up to 2000 application pods`)

File: deploy/docs/Performance.md (hunk 1 above)
Comment (perk-sumo) [reply deploy/docs/Performance.md:22]: Hm that refers to the EBS volume size on AWS I believe.

File: deploy/docs/Performance.md (hunk 2 above)
Comment (perk-sumo) [reply deploy/docs/Performance.md:24]:
```suggestion
## Up to 500 application pods

```

**Review by perk-sumo — COMMENTED** [review]
I think we should add information about the cluster configuration as well (which EC2 instances were used, disk size (125GiB), disks type (GP2), IOPS (125) etc.

If that changed between the cluster sizes we should add this info as well.

File: deploy/docs/Performance.md (hunk 1 above)
Comment (pmalek-sumo) [reply deploy/docs/Performance.md:22]: Then IMHO we should rephrase it to e.g. 

```suggestion
6. For our log generating test application pods, we found that changing the IOPS to 375 and the volume size to 125GiB improved stability.
```

**Review by pmalek-sumo — APPROVED** [review]

File: deploy/docs/Performance.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Performance
+
+For larger or more volatile loads, we recommend [enabling Fluentd autoscaling](./Best_Practices.md#Fluentd-Autoscaling), as this will allow Fluentd to automatically scale to support your data volume. However, the following recommendations and corresponding examples will help you get an idea of the resources required to run collection on your cluster.
+
+## Recommendations
+
+1. At least **8 Fluentd-logs** pods per **1 TB/day** of logs.
+2. At least **4 Fluentd-metrics** pods per **120k DPM** of metrics.
```
Comment (justrelax19) [inline deploy/docs/Performance.md:8]: Hey @samjsong , have we written 4 Fluentd-metrics by considering the scrape interval as well? Because I want to understand the numbers 120k and 4 fluentd metrics.
AFAIR, one fluentd-metrics is able to support around 60K DPM, so if we are considering the scrape interval of 30sec, then this 4 fluentd is fine but if now, can you tell me the number calculation of 4 fluentd metrics and 120K DPM

File: deploy/docs/Performance.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      remoteTimeout: 1s
+    ```
+5. For clusters with 2000 application pods, we found that the **Fluentd-events** pod had to be given a 1 GiB memory limit to accommodate the increased events load. If you find that the **Fluentd-events** pod is being OOMKilled, please increase the memory limits and requests accordingly.
+6. For our log generating test application pods, we found that increasing the IOPS to 300 minimum improved stability.
+
+### Up to 500 application pods
+
+Our test cluster had 70 nodes (AWS m5a.2xlarge instances).
+
+We used 125 GiB GP2 volumes, which allowed for IOPS of 375. In addition we attached a 300GiB volume to the Prometheus pod.
```
Comment (justrelax19) [inline deploy/docs/Performance.md:28]: We can mention for the point `In addition we attached a 300GiB volume to the Prometheus pod` that, since we were not forwarding the metrics to sumo and we were analyzing the metrics in grafana, so we need to store the metrics somewhere and therefore we used 300GiB volume

File: deploy/docs/Performance.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        regex: node-exporter;(?:node_load1|node_load5|node_load15|node_cpu_seconds_total)
+        sourceLabels: [job, __name__]
+      remoteTimeout: 1s
+    ```
+5. For clusters with 2000 application pods, we found that the **Fluentd-events** pod had to be given a 1 GiB memory limit to accommodate the increased events load. If you find that the **Fluentd-events** pod is being OOMKilled, please increase the memory limits and requests accordingly.
+6. For our log generating test application pods, we found that increasing the IOPS to 300 minimum improved stability.
+
+### Up to 500 application pods
+
+Our test cluster had 70 nodes (AWS m5a.2xlarge instances).
```
Comment (justrelax19) [inline deploy/docs/Performance.md:26]: So, out of the 70 nodes, the fluentds were running on AWS m4.large nodes.
m5a.2xlarge instances was for the prometheus and for the application pods .

IIn 500 and 2000 pods, the fluentds and k8s collection pods were m4.large instances only.

File: deploy/docs/Performance.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# Performance
+
+For larger or more volatile loads, we recommend [enabling Fluentd autoscaling](./Best_Practices.md#Fluentd-Autoscaling), as this will allow Fluentd to automatically scale to support your data volume. However, the following recommendations and corresponding examples will help you get an idea of the resources required to run collection on your cluster.
+
+## Recommendations
+
+1. At least **8 Fluentd-logs** pods per **1 TB/day** of logs.
+2. At least **4 Fluentd-metrics** pods per **120k DPM** of metrics.
+3. The Prometheus pod will use on average **2GiB** memory per **120k DPM**; however in our experience this has gone up to **5GiB**, so we recommend allocating ample memory resources for the Prometheus pod if you wish to collect a high volume of metrics for a larger cluster.
+4. For clusters with 500 application pods or greater, we found the following configuration changes in the [`values.yaml`](./../helm/sumologic/values.yaml) file to lead to a more stable experience:
```
Comment (justrelax19) [inline deploy/docs/Performance.md:10]: This is applicable for 2000 application pods as well

File: deploy/docs/Performance.md (hunk 3 above)
Comment (samjsong) [reply deploy/docs/Performance.md:8]: Right, since from Sumo backend perspective the customer is billed for 120k DPM rather than for 60k DPM at 30s scrape interval, I thought it would make it easier to understand if we kept just one number for the DPM

File: deploy/docs/Performance.md (hunk 6 above)
Comment (samjsong) [reply deploy/docs/Performance.md:10]: Agreed, I think 500 application pods or greater" should be enough to cover both of the tested cases

File: deploy/docs/Performance.md (hunk 5 above)
Comment (samjsong) [reply deploy/docs/Performance.md:26]: That's true, my opinion was that we could stick with the larger node size for simplicity but I agree we can be more accurate - I will make the change

File: deploy/docs/Performance.md (hunk 4 above)
Comment (samjsong) [reply deploy/docs/Performance.md:28]: Ah, that's a good point. In this case I think we can actually just remove this point, because it will not be needed for the customer since they will be sending their metrics to Sumo.

File: deploy/docs/Performance.md (hunk 6 above)
Comment (justrelax19) [reply deploy/docs/Performance.md:10]: Correct.

**Review by vsinghal13 — APPROVED** [review]
